### ISSUE TITLE
Docker tag :latest has older version than :v3.2.2

### ISSUE THREAD

**Issue by BlacCello** [mvdan/sh#662]
I tried to integrate shfmt into our ci pipeline with the provided docker image on dockerhub.

However, there is a missmatch between the versions and I get different formatting results.

```bash
$ docker run --rm -v "$(pwd):/mnt" mvdan/shfmt:latest --version
v3.2.0-29-g31271816

$docker run --rm -v "$(pwd):/mnt" mvdan/shfmt:v3.2.2 --version
v3.2.2
```
(The formatting differs in my case for a backslash behind && operator, followed by a newline. 
The v3.2.2 tag gives the same result as a local install with go (3.2.2))

I'm wondering what the meaning of the "latest" tag for the docker images is and why the version starts with 3.2.0 instead of 3.2.2. If this is intended, it might make sense to introduce a new tag, e.g. latest-stable or latest-release. Otherwise, I would expect that "latest" is the same as "v3.2.2" currently.

**Comment by mvdan** [comment]
The `latest` tag follows the latest development in the default branch, `master`. And it's technically correct; it currently shows the latest commit, dbbad59b44d586c0f3d044a3820c18c41b495e2a, which includes all changes in v3.2.2 as well as newer changes for the upcoming v3.3:

```
$ docker run --rm mvdan/shfmt:latest --version
Unable to find image 'mvdan/shfmt:latest' locally
latest: Pulling from mvdan/shfmt
3cea36106fa2: Pull complete 
Digest: sha256:bdca5a6640ed432effedc60b6c720cd706f4bd4344a20715e17dc80e69255806
Status: Downloaded newer image for mvdan/shfmt:latest
v3.2.0-32-gdbbad59b
```

The confusing part here is that, as per semver, master is behind the v3.2.2 tag. This is partly my fault, since I've been tagging bugfix releases off a forked branch, so then Go has to assume that master is older. See https://github.com/golang/go/issues/38985#issuecomment-626925765 for more details. I'll fix that shortly by tagging a v3.3 pre-release in master.

**Comment by mvdan** [comment]
Fixed now, with https://github.com/mvdan/sh/releases/tag/v3.3.0-0.dev:

```
$ docker run --rm mvdan/shfmt:latest --version
Unable to find image 'mvdan/shfmt:latest' locally
latest: Pulling from mvdan/shfmt
0f1463aa5a3d: Pull complete 
Digest: sha256:abf75a50be99bed27dda9a36fdd22b89dcaf6c90c953d2e07d29b87344143d4c
Status: Downloaded newer image for mvdan/shfmt:latest
v3.3.0-0.dev-34-g56c9918c
```
